webui/react/src/components/TagList.tsx

```diff
@@ -0,0 +1,144 @@
+import { PlusOutlined } from '@ant-design/icons';
+import { Input, Tag, Tooltip } from 'antd';
+import React, { useCallback, useRef, useState } from 'react';
+
+import { CommonProps } from 'types';
+
+import css from './TagList.module.scss';
+
+interface Props {
+  tags: string[];
+  setTags: (tags: string[]) => void;
+  className?: string;
+}
+
+const EditableTagList: React.FC<Props> = ({ tags, setTags, className }: Props) => {
+  const initialState = {
+    editInputIndex: -1,
+    editInputValue: '',
+    inputValue: '',
+    inputVisible: false,
+  };
+  const [ state, setState ] = useState(initialState);
+  const inputRef = useRef<Input>(null);
+  const editInputRef = useRef<Input>(null);
+
+  const handleClose = useCallback(removedTag => {
+    const newTags = tags.filter(tag => tag !== removedTag);
+    setTags(newTags);
+  }, [ tags, setTags ]);
+
+  const showInput = (e: React.MouseEvent) => {
+    setState(state => ({ ...state, inputVisible: true }));
+    inputRef.current?.focus();
```

Comment:
**suggestion:** this focus doesn't seem to work for me. I think it's because there is a delay on when the input is actually available after `inputVisible` state updates. This is a bit hacky (I don't know if there is a better way), but putting a `setTimeout` around the focus line.
```
setTimeout(() => inputRef.current?.focus(), 0);
```

Reply:
you're right it doesn't! it's because it takes some time to inputVisible state to result in the element being visible. I added a useEffect hook and moved the focus there.
```
  useEffect(() => {
    if (state.inputVisible) inputRef.current?.focus();
  }, [ state.inputVisible ]);
```

Reply:
nice way better than a timeout